core-default/src/main/java/me/aurium/beetle/defaults/task/CommonTaskRunner.java

```
@@ -0,0 +1,48 @@
+package me.aurium.beetle.defaults.task;
+
+import me.aurium.beetle.api.task.SyncQueue;
+import me.aurium.beetle.api.task.TaskRunner;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+import java.util.function.Supplier;
+
+public class CommonTaskRunner implements TaskRunner {
+
+    private final Executor asyncQueue;
+    private final Executor syncQueue;
+
+    public CommonTaskRunner(Executor syncQueue, Executor asyncQueue) {
+        this.asyncQueue = asyncQueue;
+        this.syncQueue = syncQueue;
+    }
+
+    @Override
+    public void executeSync(Runnable runnable) {
+        supplySync(() -> {
+            runnable.run();
+
+            return null;
+        });
+    }
+
+    @Override
+    public void executeAsync(Runnable runnable) {
+        supplyAsync(() -> {
```

Same issue exists here with swallowed exceptions

---

resolved